app/src/main/java/unipd/se18/ocrcamera/GalleryActivity.java

```diff
@@ -0,0 +1,385 @@
+        }
+
+        /**
+         * Action triggered when I click the delete icon in the actionBar
+         * Method called from onOptionsItemSelected
+         */
+        private void deleteCurrentPhoto()
+        {
+            try {
+                //Delete the actual image
+                GalleryManager.deleteImage(photoInfos);
+
+                //Close current fragment
+                closeDetailFragment();
+
+            } catch (IOException e) {
+                //Manage the exaption with a dialog
+                showDeletingErrorDialog();
+            }
+        }
+
+        /**
+         * Method for closing the current fragment
+         */
+        private void closeDetailFragment()
+        {
+            getFragmentManager().popBackStack();
+        }
+
+
+        /**
+         * Show a simple error message when something occur during the deletion of an image
+         */
+        private void showDeletingErrorDialog()
+        {
+            AlertDialog.Builder builder;
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+                builder = new AlertDialog.Builder(getContext(), android.R.style.Theme_Material_Dialog_Alert);
+            } else {
+                builder = new AlertDialog.Builder(getContext());
+            }
+            builder.setTitle("Error")
+                    .setMessage("Error while deleting the photo. Retry")
+                    .setPositiveButton(android.R.string.yes, new DialogInterface.OnClickListener() {
+                        public void onClick(DialogInterface dialog, int which) {
+                            //I'm not performing any action because this allert is just for provide information
+                        }
+                    })
+                    .setIcon(android.R.drawable.ic_dialog_alert)
+                    .show();
+        }
+    }
+
+
+    /*********************************************/
+    /******************PERMISSIONS****************/
+    /*********************************************/
+
+
+    /**
+     * Verify if the user granted the permission
+     */
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults)
+    {
+        if(requestCode==REQUEST_PERMISSION_CODE)
+        {
+            // If request is cancelled, the result arrays are empty.
+            if (grantResults.length > 0 && grantResults[0] != PackageManager.PERMISSION_GRANTED) {
+                // permission denied
+                showPermissionErrorDialog();
+            }
+            else
+            {
+                loadHomeFragment();
+            }
+        }
+    }
+
+    /**
+     * Verify if the app has the storage permission
+     * @return boolean of the current status (before asking the permission)
+     * if its false this prevent the app to load the fragment that uses the internal storage
+     */
+    public boolean verifyStoragePermission()
```

Comment:
The "Manifest.permission.WRITE_EXTERNAL_STORAGE" is for having the permission for deletion, right?
Does it need also the "Manifest.permission.READ_EXTERNAL_STORAGE" for viewing the pics?

Reply:
Yes WRITE_EXTERNAL_STORAGE is for deleting the images and I don't need to ask for READ_EXTERNAL_STORAGE because android.permission.WRITE_EXTERNAL_STORAGE will automatically add android.permission.READ_EXTERNAL_STORAGE to your APK at build time